File: problems/atbash/atbash.cs
using System;

namespace Atbash
{
    class Program
    {

        const string letters = "abcdefghijklmnopqrstuvwxyz";

        static void Main(string[] args)
        {
            Console.WriteLine(_Cipher("irk"));
            Console.WriteLine(_Cipher("low"));
            Console.WriteLine(_Cipher("hob"));
            Console.WriteLine(_Cipher("hold"));
            Console.ReadKey();

        }

        private static string _Cipher(string text)
        {
            string cipheredText = "";
            int index;
            foreach (var c in text)
            {
                index = letters.IndexOf(c);
                cipheredText += letters.Substring(25 - (index), 1);
            }
            return cipheredText;
        }

    }
}

File: problems/atbash/atbash.java
import java.util.stream.Collectors;
import java.util.stream.IntStream;

public class AtBash {

    private static char[] letters = IntStream.rangeClosed('a', 'z')
        .mapToObj(c -> "" + (char) c).collect(Collectors.joining()).toCharArray();

    private static String cipher(String text) {
        StringBuilder cipheredText = new StringBuilder();
        int index;
        for (char c : text.toCharArray()) {
            index = IntStream.range(0, letters.length)
                .filter(i -> letters[i] == Character.toLowerCase(c))
                .findFirst()
                .orElse(-1);
            cipheredText.append(Character.isLowerCase(c) ? Character.toUpperCase(letters[25 - index]) : Character.toLowerCase((letters[25 - index])));
        }
        return cipheredText.toString();
    }

    public static void main(String[] args) {
        System.out.println(cipher("irk"));
        System.out.println(cipher("LOW"));
        System.out.println(cipher("hob"));
        System.out.println(cipher("hold"));
        System.out.println(cipher("AaZz"));
    }
}
